tests/integration/pilot/analyze_test.go

```diff
@@ -387,9 +398,24 @@ func istioctlWithStderr(t *testing.T, i istioctl.Instance, ns string, useKube bo
 	return i.Invoke(args)
 }
 
-func applyFileOrFail(t *testing.T, ns, filename string) {
-	t.Helper()
-	if err := cluster.ApplyYAMLFiles(ns, filename); err != nil {
-		t.Fatal(err)
+func applyFileOrFail(ctx framework.TestContext, ns, filename string) interface {
+	Delete()
+} {
+	ctx.Helper()
+	if err := ctx.Clusters().Default().ApplyYAMLFiles(ns, filename); err != nil {
+		ctx.Fatal(err)
 	}
+	return &cleanup{
```

Comment:
Is it possible to use it here?
https://github.com/istio/istio/blob/a9a5d5ce347baa6fbc19ccaeebc9daa92f2fec19/pkg/test/failer.go#L30

Reply:
Copying a slack comment here for longevity: 

> what do you think the behavior of --nocleanup in integration test should be for config applied during a test? Currently its inconsistent due to different usage patterns (defer vs WhenDone vs Cleanup).
> We can respect the flag, but that means that you cannot run multiple tests (conflicting resources will be created and still remain). Or we can ignore it and always delete yaml, but it makes the flag a bit confusing. Or we can add a third flag. Any thoughts?

Basically we can probably do something better holistically here